packages/amesos2/src/Amesos2_Tacho_decl.hpp

```diff
@@ -96,10 +98,20 @@ class TachoSolver : public SolverCore<Amesos2::TachoSolver, Matrix, Vector>
   typedef Tacho::ordinal_type                                  ordinal_type;
   typedef Tacho::size_type                                        size_type;
   typedef Kokkos::DefaultHostExecutionSpace                   HostSpaceType;
-  typedef Kokkos::TaskScheduler<HostSpaceType>                SchedulerType;
-  typedef Kokkos::View<size_type*,HostSpaceType>            size_type_array;
-  typedef Kokkos::View<ordinal_type*,HostSpaceType>      ordinal_type_array;
-  typedef Kokkos::View<tacho_type*, HostSpaceType>         value_type_array;
+
+  typedef Kokkos::DefaultExecutionSpace                      DeviceSpaceType;
+
+
+  typedef Kokkos::TaskScheduler<DeviceSpaceType>              SchedulerType;
+
+  typedef Kokkos::View<size_type*, DeviceSpaceType>       device_size_type_array;
+  typedef Kokkos::View<ordinal_type*, DeviceSpaceType> device_ordinal_type_array;
```

Comment:
I would probably use "typename Tacho::UseThisDevice<exec_space>::device_type" with typedef. I lately include this fix as the scheduler requires uvm memory space but the other internal views do not require UVM. See, Tacho_Util.hpp

Reply:
Changed to now use Tacho::UseThisDevice.